test/e2e/storage/testsuites/provisioning.go

```diff
@@ -280,6 +290,9 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
 		if pattern.VolMode == v1.PersistentVolumeBlock && !dInfo.Capabilities[storageframework.CapBlock] {
 			e2eskipper.Skipf("Driver %q does not support block volumes - skipping", dInfo.Name)
 		}
+		if strings.HasPrefix(dInfo.Name, "csi-gcepd") {
```

Comment:
Adding driver specific checks here does not look right. Instead we should use a skip string for the parallel test, in the tester source. e.g pd csi [repository](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/tree/master/test/k8s-integration).  The skips in the test suite are only based on capabilties, we may also explore adding a Parallel cloning capability. Thoughts?

Reply:
I think the skip string makes more sense because we could be the only storage driver provider that doesn't support parallel volume cloning, so then other providers would have to remember to turn on that capability to run the parallel tests. And we would probably want to update the Volume Cloning section of the CSI Developer Documentation to explain that.

Reply:
Especially with us starting to run pd csi tests from cloud-provider-gcp, I'm not sure it's really maintainable to rely on test driver configs to keep track of which tests are passing. But I'm interested to hear what the consensus is.

Another possibility is to increase the timeouts. The issue is that GCE only supports one clone per device per 30 seconds, or something like that, so this test would pass if the timeouts were increased.

Reply:
I don't like driver specific workaround in tests too.

As short-term fix, increasing a timeout sounds OK. Mid term, it would be cool if each driver could provide its own timeouts, esp. when running the [tests outside of k/k](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/external/README.md).

Long term, I think we have a broader problem with CSI. [It expects that](https://github.com/container-storage-interface/spec/blob/master/spec.md#concurrency)
> In general the Cluster Orchestrator (CO) is responsible for ensuring that there is no more than one call “in-flight” per volume at a given time. However, in some circumstances, the CO MAY lose state (for example when the CO crashes and restarts), and MAY issue multiple calls simultaneously for the same volume.

Kubernetes sort of ignores this paragraph. It deletes volume while detaching them, it takes snapshot at any time, it can resize at any time and here it clones volume while detaching. I don't have a smart solution in my head, Kubernetes (and CSI sidecars) is a dynamic system and adding synchronization will hurt its performance. Anyway, it does not need to be solved here.

Reply:
ah, I like the idea of passing per-driver timeouts.

@amacaskill can your look into added a [custom timeout](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/external/external.go#L134) for cloning?

Reply:
That's an interesting point about the one in-flight requirement. We've actually run into that in attach-detach for PD CSI and are solving that in the driver. I wonder if we should just update the spec as it seems to be a dead letter.

@saikat-royc

Reply:
I added custom timeout for cloning. Let me know if there is anything else I should change.

Reply:
I would expect that only the new cloning timeout to be in this PR and then you could bump the timeout in GCP PD test job.

Reply:
Yes I changed (this line)[https://github.com/kubernetes/kubernetes/blob/4e7d4d38566cfc93f9e4291d18382ee13b468343/test/e2e/storage/testsuites/provisioning.go#L431]  to be the new custom timeout and then in the  GCP PD test job I increased the time from 5m (what it was before) to 8m and now everything is passing consistently.